src/toil/jobStores/utils.py

```diff
@@ -109,3 +110,33 @@ def single_attempt():
             yield
 
         yield single_attempt()
+
+default_delays = (0, 1, 1, 4, 16, 64)
+default_timeout = 300
+
+
+def retry_http(delays=default_delays, timeout=default_timeout, predicate=None):
+    """
+    >>> i = 0
+    >>> for attempt in retry_http():
+    ...     with attempt:
+    ...         i += 1
+    ...         raise urllib2.HTTPError('http://www.test.com', '408', 'some message', {}, None)
+    Traceback (most recent call last):
+    ...
+    HTTPError: HTTP Error 408: some message
+    >>> print i
+    3
+    >>> for attempt in retry_http():
```

Comment:
1. [http://jenkins.cgcloud.info/job/toil-pull-requests/1532/testReport/src.toil.jobStores/utils/toil_jobStores_utils_retry_http/](http://jenkins.cgcloud.info/job/toil-pull-requests/1532/testReport/src.toil.jobStores/utils/toil_jobStores_utils_retry_http/)

2. The first failure would trigger the second even if it passes. I originally figured this was feature not a bug since if the first one fails its extremely likely the second will fail.  But this doesn't accomplish much since you correctly pointed out that we get no extra coverage.